client/components/domains/registrant-extra-info/fr-form.jsx

```diff
@@ -0,0 +1,301 @@
+/**
+ * External dependencies
+ */
+import React, { PropTypes } from 'react';
+import { isEqual, noop } from 'lodash';
+import { localize } from 'i18n-calypso';
+import moment from 'moment';
+import debugFactory from 'debug';
+
+/**
+ * Internal dependencies
+ */
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import FormLegend from 'components/forms/form-legend';
+import FormRadio from 'components/forms/form-radio';
+import FormCountrySelect from 'components/forms/form-country-select';
+import FormTextInput from 'components/forms/form-text-input';
+import FormSettingExplanation from 'components/forms/form-setting-explanation';
+import { forDomainRegistrations as countriesListForDomainRegistrations } from 'lib/countries-list';
+
+const debug = debugFactory( 'calypso:domains:registrant-extra-info' );
+
+const countriesList = countriesListForDomainRegistrations();
+
+class RegistrantExtraInfoForm extends React.PureComponent {
+	static propTypes = {
+		isVisible: PropTypes.bool,
+		onSubmit: PropTypes.func,
+		countriesList: PropTypes.object,
+		onStateChanged: PropTypes.func, // Just until we can reduxify the contact details
+	}
+
+	static defaultProps = {
+		isVisible: true,
+		isProbablyOrganization: false,
+		onSubmit: noop,
+		countriesList: countriesList,
+	}
+
+	constructor( props ) {
+		super( props );
+	}
+
+	componentWillMount() {
+		this.setState( {
+			registrantType: this.props.isProbablyOrganization
+				? 'organization' : 'individual',
+		} );
+	}
+
+	componentDidMount() {
+		// Make sure that we send an an initial update in case the user submits
+		// without changing anything.
+		this.props.onStateChange &&
+			this.props.onStateChange( this.state );
+	}
+
+	getRelevantFields( state ) {
```

Comment:
> @hewsut: what does `getRelevantFields` do? I mean, I understand what it does inside the function. But I don’t see it called anywhere?

Nice catch! It captures the way the .FR fields depend on each other. I was using it on the state before passing it out at one point, but now I'm not sure - maybe we want to save everything to the database? At any rate, we will need it to handle validation (we don't care about errors in the the organization fields for individuals), but it's not being used right now.  I should probably pull it out, but I don't think I will just yet unless someone objects strongly.